### PR TITLE
roles-tag-and-release.sh - edit CHANGELOG.md

### PR DESCRIPTION
**Pull request by nhosoi**
When editing CHANGELOG.md, if the file starts with "# Changelog" or
"Changelog/n===...", keep them there.

### CODE REVIEW

File: roles-tag-and-release.sh [hunk 1]
```diff
@@ -106,7 +106,21 @@ if [ "$skip" = false ]; then
         ${EDITOR:-vi} "$rel_notes_file"
         if [ -f CHANGELOG.md ]; then
             read -r -p "Edit CHANGELOG.md - press Enter to continue"
-            cat "$rel_notes_file" CHANGELOG.md > .tmp-changelog
+            pat="=*"
+            if [[ $( head -n 1 CHANGELOG.md ) == "# Changelog" ]]; then
```
Comment (richm) [inline roles-tag-and-release.sh:110]: Why do we have two different styles of changelog file?  We should have only 1

Comment (nhosoi) [reply roles-tag-and-release.sh:110]: If we don't have to support
```
Changelog
=========
```
I'm more than happy! I'm eliminating it and just support `# Changelog`.
Thanks!

Comment (richm) [reply roles-tag-and-release.sh:109]: note that the network role is the only role that currently has a CHANGELOG and it uses
```
Changelog
=========
```
What if we only use that form?  Is there somewhere else that uses the `# Changelog` form?

Comment (nhosoi) [reply roles-tag-and-release.sh:109]: Well, I already generated the initial CHANGELOG.md for most of the roles using `github-changelog-generator`, which generates `# Changelog` form...

Comment (richm) [reply roles-tag-and-release.sh:109]: ok - I think the network role shouldn't mind changing the `====` form to the `# Changelog` form - either way is fine

Comment (nhosoi) [reply roles-tag-and-release.sh:109]: Fingers crossed... May I ask them if they don't mind?

Comment (richm) [reply roles-tag-and-release.sh:109]: > Fingers crossed... May I ask them if they don't mind?

Sure

File: roles-tag-and-release.sh [hunk 2]
```diff
@@ -106,7 +106,17 @@ if [ "$skip" = false ]; then
         ${EDITOR:-vi} "$rel_notes_file"
         if [ -f CHANGELOG.md ]; then
             read -r -p "Edit CHANGELOG.md - press Enter to continue"
-            cat "$rel_notes_file" CHANGELOG.md > .tmp-changelog
+			pat="=*"
```
Comment (richm) [inline roles-tag-and-release.sh:109]:
```bash
  myheader="Changelog
========="
  clheader=$(head -2 CHANGELOG.md)
  if [ "$myheader" = "$clheader" ]; then
    echo "$clheader" > .tmp-changelog
    cat "$rel_notes_file" >> .tmp-changelog
    tail -n +3 CHANGELOG.md >> .tmp-changelog
  else
    echo WARNING: Changelog header "$clheader"
    echo not in expected format "$myheader"
    cat "$rel_notes_file" CHANGELOG.md > .tmp-changelog
  fi
```

Comment (nhosoi) [reply roles-tag-and-release.sh:109]: Thanks, @richm. If we are more conscious about the markdown format, `roles-tag-and-release.sh` generates `## [VERSION] - DATE` and stores in `$rel_notes_file`. We still don't know to run `markdownlint.yml` action in each role, but considering the rules set for the network role, we'd like to have `[1.9.4] - 2022-06-28` in the Setext style. Do you think it's possible? (please note that it's ok to have `### New features` and `### Bug fixes`.)

Example of `$rel_notes_file`.
```
## [1.9.4] - 2022-06-28

### New features

- use gather_facts: true in tests_include_vars_from_parent
- support ansible-core-2.13

### Bug fixes

- make min_ansible_version a string in meta/main.yml
```

Comment (nhosoi) [reply roles-tag-and-release.sh:109]: BTW, it's not a good idea to replace the ATX format with Setext before inserting "$rel_notes_file" like this?
```
sed -i -e "s/^# Changelog/Changelog\n=========/" \
       -e "s/^## \(.*\)/\1\n--------------------/" CHANGELOG.md
```
Is it too brutal?

**Review by richm — APPROVED** [review]
lgtm

**Comment by nhosoi** [comment]
> lgtm

Thank you, @richm. I made a small change to support `v##.##.##` style for `sshd`.

File: roles-tag-and-release.sh [hunk 3]
```diff
@@ -56,9 +57,13 @@ fi
...
+# special case for network and sshd
 case "$latest_tag" in
-v*) latest_tag="${latest_tag//v}" ;;
+v*)
+    if [ "$repo" != "ansible-sshd" ]; then
+        latest_tag="${latest_tag//v}"
+    fi
+    ;;
 esac
 skip=false
```
Comment (richm) [inline roles-tag-and-release.sh:68]: add
```
# shellcheck disable=SC2129
```
to make shellcheck tests pass

**Review by richm — APPROVED** [review]
lgtm

**Pull request merged**